java/org/apache/coyote/http11/Http11InputBuffer.java

```diff
@@ -550,10 +550,15 @@ boolean parseRequestLine(boolean keptAlive, int connectionTimeout, int keepAlive
                 prevChr = chr;
                 chr = byteBuffer.get();
                 if (chr == Constants.CR) {
-                    // Possible end of request line. Need LF next.
+                    // Possible end of request line. Need LF next else invalid.
                 } else if (prevChr == Constants.CR && chr == Constants.LF) {
+                    // CRLF is the standard line terminator
                     end = pos - 1;
                     parsingRequestLineEol = true;
+                } else if (chr == Constants.LF) {
```

Comment:
Is it worth "fingerprinting" a message to determine if it is self-consistent? That is, a message is allowable with LF-terminated request/headers but _only_ if *all* of its headers and the request line are terminated with LF (only)? Or is that being unnecessarily picky and complicating the code?

Reply:
It would make the code a little more complex and there is nothing in the spec to suggest that self-consistency is required. If we were going to make the behaviour configurable I can see how this would fit into an option that allow varying degrees of LF use. Eg: none, consistent, any.
My current thinking is to essentially hard-code the any option unless someone identifies a good reaaon not to.